### ISSUE TITLE
Incorrect ParticleIDs

### ISSUE THREAD

**Issue by JBorrow** [JBorrow/lagrangian-transfer#1]
Currently the code tries to just match ParticleIDs automatically. This clearly doesn't work -- there are some bits modified by GIZMO when it forms stars.

- [x] Figure out what is going on with the IDs
- [x] Test on small sub-set of data
- [x] Implement into the code

**Comment by JBorrow** [comment]
It looks like star particles that "eat" a whole gas particle are fine. They stick with their original ID. Otherwise, there is some bit manipulation on the bits just above the maximum particle ID: this is somewhat dangerous, but I guess it's what we've got!

My suggestion for now (still yet to implement it) is just to ignore all bits higher than the highest bit occupied when looking at the maximum particle ID. We don't really care how many star particles we formed, as we look at the mass (and I would hope that mass is conserved) anyway.

In the 25 Mpc 256^3 box, there are 16777216 (`00000001000000000000000000000000`) particles, with a maximal ID of 16777215, `00000000111111111111111111111111`.

particles that do not lie in both lists seem to have their higher bits played with. Here is a small list of them:

```
# Old max ID for comparison
00000000111111111111111111111111
# New Ids
00000010000000000000000000000000
00000010000000000000000000000001
00000010000000000000000000000010
00000010000000000000000000000011
00000010000000000000000000000100
00000010000000000000000000000101
00000010000000000000000000000110
00000010000000000000000000000111
00000010000000000000000000001000
00000010000000000000000000001001
00000010000000000000000000001010
00000010000000000000000000001011
00000010000000000000000000001101
00000010000000000000000000001111
00000010000000000000000000010000
00000010000000000000000000010001
00000010000000000000000000010010
00000010000000000000000000010011
00000010000000000000000000010100
00000010000000000000000000010101
00000010000000000000000000010110
00000010000000000000000000010111
00000010000000000000000000011000
00000010000000000000000000011001
00000010000000000000000000011010
00000010000000000000000000011011
00000010000000000000000000011100
00000010000000000000000000011101
00000010000000000000000000011110
00000010000000000000000000100000
00000010000000000000000000100001
00000010000000000000000000100011
00000010000000000000000000100100
00000010000000000000000000100101
00000010000000000000000000100110
00000010000000000000000000100111
00000010000000000000000000101000
00000010000000000000000000101001
00000010000000000000000000101010
00000010000000000000000000101011
00000010000000000000000000101100
00000010000000000000000000101101
00000010000000000000000000101110
00000010000000000000000000101111
00000010000000000000000000110000
00000010000000000000000000110001
00000010000000000000000000110011
00000010000000000000000000110100
00000010000000000000000000110101
00000010000000000000000000110110
00000010000000000000000000111000
00000010000000000000000000111001
00000010000000000000000000111010
00000010000000000000000000111011
00000010000000000000000000111100
00000010000000000000000000111101
00000010000000000000000000111110
00000010000000000000000000111111
00000010000000000000000001000000
00000010000000000000000001000001
00000010000000000000000001000010
00000010000000000000000001000100
00000010000000000000000001000101
00000010000000000000000001000110
00000010000000000000000001000111
00000010000000000000000001001000
00000010000000000000000001001001
00000010000000000000000001001010
00000010000000000000000001001011
00000010000000000000000001001101
00000010000000000000000001001110
00000010000000000000000001001111
00000010000000000000000001010000
00000010000000000000000001010001
00000010000000000000000001010010
00000010000000000000000001010011
00000010000000000000000001010100
00000010000000000000000001010101
00000010000000000000000001010111
00000010000000000000000001011000
00000010000000000000000001011001
00000010000000000000000001011010
00000010000000000000000001011100
00000010000000000000000001011101
00000010000000000000000001011110
00000010000000000000000001011111
00000010000000000000000001100001
00000010000000000000000001100010
00000010000000000000000001100100
00000010000000000000000001100101
00000010000000000000000001100110
00000010000000000000000001100111
00000010000000000000000001101000
00000010000000000000000001101001
00000010000000000000000001101010
00000010000000000000000001101100
00000010000000000000000001101101
00000010000000000000000001101110
00000010000000000000000001101111
00000010000000000000000001110010
00000010000000000000000001110011
00000010000000000000000001110100
00000010000000000000000001110101
00000010000000000000000001110110
00000010000000000000000001111000
00000010000000000000000001111010
00000010000000000000000001111011
00000010000000000000000001111100
00000010000000000000000001111101
00000010000000000000000001111110
```

I'll keep this code self-contained so it can be shared more easily with others who will be using MUFASA.

**Comment by JBorrow** [comment]
Looks like the best way to implement this, as we don't care about the 'repeated IDs', is to just truncate. We can do this by taking the modulo with respect to the total number of IDs +1, i.e. `ids %= max_id + 1`. This should just give us the IDs without any star formation.

**Comment by JBorrow** [comment]
So it looks like this was fixed in 38bbf6e74eadbe07664f93ebbccdb468f0ba7b51 and 535ef3e5013ad6ac231001ce73499f6a39204e66.
